glean-core/ios/Glean/Glean.swift

```diff
@@ -39,6 +39,10 @@ public class Glean {
 
     private let logger = Logger(tag: Constants.logTag)
 
+    // Cache variable for checking if running in main process.  Also used to override for tests in
+    // order to simulate not running in the main process.  DO NOT SET EXCEPT IN TESTS!
```

Comment:
I'm ... not super happy that this is potentially exposed to users and all that is stopping them to touch this is a comment.
Do we really need a test for this? If not this could be private and unchangeable after first setting.

If we do, couldn't we make this "internal" and then it's only available if Glean is imported `@testable`?

Reply:
Don't forget, `internal` is the default access for variables in Swift.  This would only be exposed if it were explicitly `public`

Reply:
I originally had it as private but, as you noticed, it was easier to test this way.  I'm okay dropping the test and making this explicitly private too, but having test coverage felt useful...  (looks over shoulder for @Dexterp37)

Reply:
uuuh, I constantly forget about `internal` being the default. That means under normal use one could already not read or write this field, right?

Reply:
Correct, only those that import Glean as `@testable` should be able to access it.

Reply:
That makes me less woried then